x-pack/plugin/fleet/src/main/java/org/elasticsearch/xpack/fleet/Fleet.java

```diff
@@ -300,6 +301,7 @@ public void cleanUpFeature(ClusterService clusterService, Client client, ActionL
         if (dataStreamDescriptors.isEmpty() == false) {
             try {
                 Request request = new Request(
+                    TimeValue.THIRTY_SECONDS /* TODO should we wait longer? */,
```

Comment:
Unsure about this timeout tbh, I'm leaving it at 30s but at least calling it out as such here.